rmw_cyclonedds_cpp/CMakeLists.txt

```diff
@@ -33,7 +33,16 @@ find_package(tracetools REQUIRED)
 
 #find_package(cyclonedds_cmake_module REQUIRED)
 find_package(CycloneDDS QUIET CONFIG)
-if(NOT CycloneDDS_FOUND)
+if(CycloneDDS_FOUND)
+  # Support for shared memory in RMW depends on support for shared memory being compiled in
+  # Cyclone DDS and iceoryx_binding_c being available
+  get_target_property(cyclonedds_has_shm CycloneDDS::ddsc SHM_SUPPORT_IS_AVAILABLE)
+  if(cyclonedds_has_shm)
+    find_package(iceoryx_binding_c REQUIRED)
```

Comment:
This will likely not work on the buildfarm as-is.  If you want to do this, you'll have to add a package.xml dependency to iceoryx.

Reply:
Thanks for pointing this out, I added `iceoryx_binding_c` dependency to `Package.xml`